src/Analyzers/Core/Analyzers/UseSystemHashCode/Analyzer.OperationDeconstructor.cs

```diff
@@ -108,7 +109,9 @@ public bool TryAddHashedSymbol(IOperation value, bool seenHash)
                             if (binary.OperatorKind == BinaryOperatorKind.Equals)
                             {
                                 // (StringProperty == null ? 0 : StringProperty.GetHashCode())
-                                return TryAddHashedSymbol(conditional.WhenFalse, seenHash: true);
+                                Debug.Assert(conditional.WhenFalse is not null);
+                                // Need the suppression for codestyle, which doesn't respect the above assertion
```

Comment:
There should be a bug tracking this or we should switch to a code style that works with nullable .

Reply:
As in the `CodeStyle` IDE project. It doesn't target .NET 5, so it doesn't have an annotated version of `Debug.Assert`

Reply:
Why not `RoslynDebug.Assert`? That is what we use every where else

Reply:
I could, but I'd been trying to avoid that API. I guess for this space I should use it though.